controllers/utils/fieldindexer.go

```diff
@@ -18,48 +18,37 @@ package util
 
 import (
 	"context"
-	"sync"
 
 	"github.com/openyurtio/yurt-edgex-manager/api/v1alpha1"
 	"github.com/openyurtio/yurt-edgex-manager/api/v1alpha2"
-
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
-	IndexerPathForNodepool = "spec.poolname"
+	IndexerPathForNodepoolv1 = "spec.poolname"
+	IndexerPathForNodepoolv2 = "spec.poolname"
```

Comment:
One problem is that the correct path should be `spec.poolName`. The two constants can be combined into one, because this represents the path to the index. Since v1alpha1 and v1alpha2 are both built using poolName, they do not differentiate between versions, and the names v1 and v2 are ambiguous.